frontend/src/api/api.ts

```diff
@@ -0,0 +1,203 @@
+// Still a few kinks to be ironed out - TS sees no issue but our linter does
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+// Response return is read as any by our linter for some reason
+/* eslint-disable @typescript-eslint/no-unsafe-return */
+import { AxiosResponse } from 'axios';
+import { SortDirection } from '@tanstack/react-table';
+import { CLValueParsers } from 'casper-js-sdk';
+import { formatDate, formatTimeAgo, loadConfig } from 'src/utils';
+import { createBaseApi } from './base-api';
+import { ApiData, DeployStatus } from './types';
+import { determineDeploySessionData, isValidPublicKey } from './utils';
+import { JsonDeploySession } from './missing-sdk-types';
+
+const { webServerUrl } = loadConfig();
+
+const createApi = (baseUrl: string) => {
+  const middlewareApi = createBaseApi(baseUrl);
+
+  return {
+    block: {
+      async getBlocks(tableParams: {
```

Comment:
NIT: Thoughts on having table params be optional and defaulting to an empty object?

Reply:
Yup, I like it.